pandas/tests/indexes/test_base.py

```diff
@@ -1092,25 +1058,23 @@ def test_intersection_difference(self):
             diff = idx.difference(idx)
             tm.assert_index_equal(inter, diff)
 
-    def test_is_numeric(self):
-        assert not self.dateIndex.is_numeric()
-        assert not self.strIndex.is_numeric()
-        assert self.intIndex.is_numeric()
-        assert self.floatIndex.is_numeric()
-        assert not self.catIndex.is_numeric()
-
-    def test_is_object(self):
-        assert self.strIndex.is_object()
-        assert self.boolIndex.is_object()
-        assert not self.catIndex.is_object()
-        assert not self.intIndex.is_object()
-        assert not self.dateIndex.is_object()
-        assert not self.floatIndex.is_object()
-
-    def test_is_all_dates(self):
-        assert self.dateIndex.is_all_dates
-        assert not self.strIndex.is_all_dates
-        assert not self.intIndex.is_all_dates
+    @pytest.mark.parametrize("attr,exp", [
+        ('strIndex', False), ('boolIndex', False), ('catIndex', False),
+        ('intIndex', True), ('dateIndex', False), ('floatIndex', True)])
+    def test_is_numeric(self, attr, exp):
```

Comment:
This and the two tests below it did not consistently test all of the index types so I've added where missing for better coverage. Should ultimately be replaced with fixtures but same concept will apply